allauth/utils.py

```diff
@@ -43,6 +45,11 @@ def _generate_unique_username_base(txts, regex=None):
         username = username.split('@')[0]
         username = username.strip()
         username = re.sub('\s+', '_', username)
+        # Finally, validating base username without database lookups etc.
+        try:
+            username = adapter.clean_username(username, shallow=True)
+        except ValidationError:
+            pass
         if username:
             break
```

Comment:
This is not sound... `username` is always se.. so we always break. So the break should be done above the `except` just after invoking `clean_username`